spec/features/lists_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'rails_helper'
+require_relative 'helpers'
+
+RSpec.configure do |c|
+  c.include FeatureHelpers
+end
+
+RSpec.feature 'Lists', type: :feature do
+  let(:list_name) { 'My awesome list' }
+  let(:another_list_name) { 'Another great list' }
+
+  before do
+    visit root_path
+  end
+
+  describe 'creating a list' do
+    scenario 'user creates a new list' do
+      create_list(list_name)
+
+      expect(page).to have_content list_name
+      expect(page).to have_content 'No items have yet been added'
+    end
+  end
+
+  describe 'deleting a list' do
+    before do
+      create_list(another_list_name)
+      visit root_path
+    end
+
+    scenario 'user deletes a list' do
+      list = find('li', text: another_list_name)
+      delete_link = list.all('a').last.click # TODO: fragile
```

Comment:
My HTML output looks like this:

```html
<li class="row">
   <a class="columns small-10" href="/lists/42">Katherine&#39;s awesome list</a>
   <a data-confirm="Are you sure?" class="columns small-2" data-remote="true" rel="nofollow" data-method="delete" href="/lists/42">
      <i class="fi-trash"></i>
   </a>        
</li>
```

`list` is the full `<li>`.  
To get the delete link, I need to look with `list` and find the link containing the icon (but not the icon itself).  
Is there a way to do this with Capybara selectors that I'm not seeing, or do I simply need to use XPath?

Reply:
So it's really common to add css classes that identify the link you want, and to use capybara's css selectors to find it.

Reply:
I'd have to add a CSS class to that second link just for this test, though. That's kosher?

Reply:
You don't have to add the class to any css file, for the purpose of testing only, similar to how you  might add a class to an element so that you can select it using Jquery, it could be something like class='confirm-link' on the link, that's pretty common